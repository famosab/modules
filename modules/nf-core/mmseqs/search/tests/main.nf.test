nextflow_process {

    name "Test Process MMSEQS_SEARCH"
    script "../main.nf"
    process "MMSEQS_SEARCH"

    tag "modules"
    tag "modules_nfcore"
    tag "mmseqs"
    tag "mmseqs/search"
    tag "mmseqs/createdb"

    setup {
        run("MMSEQS_CREATEDB", alias: "MMSEQS_CREATEDB_QUERY") {
            script "../../../mmseqs/createdb/main.nf"
            process {
                """
                input[0] = [ [ id:'test_query', single_end:true ], // meta map
                    file(params.test_data['sarscov2']['illumina']['contigs_fasta'], checkIfExists: true)
                ]
                """
            }
        }

        run("MMSEQS_CREATEDB", alias: "MMSEQS_CREATEDB_TARGET") {
            script ".../../../mmseqs/createdb/main.nf"
            process {
                """
                input[0] = [ [ id:'test_target', single_end:true ], // meta map
                    file(params.test_data['sarscov2']['illumina']['scaffolds_fasta'], checkIfExists: true)
                ]
                """
            }
        }
    }

    test("sarscov2 - test-search") {

        when {
            params {
                module_args   = '--remove-tmp-files 1 --search-type 2'
                module_prefix = 'test_output_search'
            }
            process {
                """
                input[0] = MMSEQS_CREATEDB_QUERY.out.db
                input[1] = MMSEQS_CREATEDB_TARGET.out.db
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out,
                    path(process.out.versions[0]).yaml
                    ).match() }
            )
        }

    }

    test("sarscov2 - test-search - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = MMSEQS_CREATEDB_QUERY.out.db
                input[1] = MMSEQS_CREATEDB_TARGET.out.db
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out,
                    path(process.out.versions[0]).yaml
                    ).match() }
            )
        }

    }

}
