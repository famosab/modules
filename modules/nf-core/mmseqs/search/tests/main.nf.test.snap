{
    "sarscov2 - test-search": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_query",
                            "single_end": true
                        },
                        [
                            "test_output_search.0:md5,e85671ec77d822e9b952110e7f87ef93",
                            "test_output_search.1:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.dbtype:md5,cbe23d9655fccd4311acdd6261c81bd8",
                            "test_output_search.index:md5,c9323312ba44d0a8bb44136d7d5b951f"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,2568126b1f8eb47999b31cd5f3b9c68e"
                ],
                "db_search": [
                    [
                        {
                            "id": "test_query",
                            "single_end": true
                        },
                        [
                            "test_output_search.0:md5,e85671ec77d822e9b952110e7f87ef93",
                            "test_output_search.1:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.dbtype:md5,cbe23d9655fccd4311acdd6261c81bd8",
                            "test_output_search.index:md5,c9323312ba44d0a8bb44136d7d5b951f"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2568126b1f8eb47999b31cd5f3b9c68e"
                ]
            },
            {
                "MMSEQS_SEARCH": {
                    "mmseqs": "15.6f452"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-15T09:28:21.271352337"
    },
    "sarscov2 - test-search - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_query",
                            "single_end": true
                        },
                        [
                            "test_output_search.0:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.1:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.2:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.3:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.4:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.5:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.6:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.7:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.8:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.9:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.dbtype:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.index:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,2568126b1f8eb47999b31cd5f3b9c68e"
                ],
                "db_search": [
                    [
                        {
                            "id": "test_query",
                            "single_end": true
                        },
                        [
                            "test_output_search.0:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.1:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.2:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.3:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.4:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.5:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.6:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.7:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.8:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.9:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.dbtype:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_output_search.index:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2568126b1f8eb47999b31cd5f3b9c68e"
                ]
            },
            {
                "MMSEQS_SEARCH": {
                    "mmseqs": "15.6f452"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-15T09:28:46.406630911"
    }
}